event/event.py

```diff
@@ -41,10 +42,39 @@ def character_gate(self):
     def characters_required(self):
         return 1
 
-    def add_reward(self, possible_types):
-        new_reward = Reward(self, possible_types)
-        self.rewards.append(new_reward)
-        return new_reward
+    def get_reward_type(self, check_info = None):
+        bit = check_info.bit
+
+        assert bit
+
+        if bit in self.args.esper_item_rewards:
+            return RewardType.ESPER | RewardType.ITEM
+
+        if bit in self.args.esper_rewards:
+            return RewardType.ESPER
+
+        if bit in self.args.item_rewards:
+            return RewardType.ITEM
+
+        return check_info.reward_types
+
+    def add_item_reward(self):
+        reward = Reward(self, RewardType.ITEM)
+        self.rewards.append(reward)
+        return reward
+
+    def add_character_reward(self):
+        reward = Reward(self, RewardType.CHARACTER)
+        self.rewards.append(reward)
+        return reward
+
+    def add_reward(self, check_info):
+        possible_types = self.get_reward_type(check_info)
```

Comment:
This is the entry point to adding a check now - It'll discern the current reward type and apply the force item/esper flags above